api_server.py

```
@@ -74,101 +81,94 @@ def format_hours(minutes: Optional[int]) -> Optional[str]:
-        curr = conn.execute(
-            f"SELECT end_time FROM {table} WHERE {col} = ? AND start_time <= ? AND end_time > ? LIMIT 1",
-            (entity_id, now, now),
-        ).fetchone()
-        if not curr:
-            return {"available": False, "duration": None, "end_time_display": None}
-
-        end_time = parse_dt(curr["end_time"])
-        duration_min = int((end_time - now).total_seconds() / 60)
+    conn = get_db()
+    curr = conn.execute(
+        f"SELECT end_time FROM {table} WHERE {col} = ? AND start_time <= ? AND end_time > ? LIMIT 1",
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)